src/test/scala/org/broadinstitute/dsde/rawls/jobexec/ExecutionServiceClusterTest.scala

```diff
@@ -0,0 +1,199 @@
+
+  // additional test data
+  val execClusterTestData = new ExecClusterTestData()
+
+  class ExecClusterTestData() extends TestData {
+    val wsName = WorkspaceName("ExecClusterTestDataNamespace", "ExecClusterTestDataName")
+    val user = RawlsUser(userInfo)
+    val ownerGroup = makeRawlsGroup("ExecClusterTestDataOwnerGroup", Set(user))
+    val workspace = Workspace(wsName.namespace, wsName.name, None, UUID.randomUUID().toString, "ExecClusterTestDataBucket", currentTime(), currentTime(), "testUser", Map.empty, Map(WorkspaceAccessLevels.Owner -> ownerGroup), Map(WorkspaceAccessLevels.Owner -> ownerGroup))
+
+    val sample1 = Entity("sample1", "Sample", Map("type" -> AttributeString("normal")))
+    val sample2 = Entity("sample2", "Sample", Map("type" -> AttributeString("normal")))
+
+    val submissionMissingExecutionKeys = Submission(subMissingExecutionKeys,testDate, testData.userOwner, "std","someMethod",sample1.toReference,
+      Seq(Workflow(Some(workflowExternalIdMissingExecutionKey),WorkflowStatuses.Submitted,testDate,sample1.toReference, testData.inputResolutions)),
+      Seq.empty[WorkflowFailure], SubmissionStatuses.Submitted)
+
+    val submissionWithExecutionKeys = Submission(subWithExecutionKeys,testDate, testData.userOwner, "std","someMethod",sample2.toReference,
+      Seq(Workflow(Some(workflowExternalIdWithExecutionKey),WorkflowStatuses.Submitted,testDate,sample2.toReference, testData.inputResolutions)),
+      Seq.empty[WorkflowFailure], SubmissionStatuses.Submitted)
+
+    override def save() = {
+      DBIO.seq(
+        rawlsUserQuery.save(user),
+        rawlsGroupQuery.save(ownerGroup),
+        workspaceQuery.save(workspace),
+        withWorkspaceContext(workspace) { context =>
+          DBIO.seq(
+            entityQuery.save(context, sample1),
+            entityQuery.save(context, sample2),
+            methodConfigurationQuery.save(context, MethodConfiguration("std", "someMethod", "Sample", Map.empty, Map.empty, Map.empty, MethodRepoMethod("std", "someMethod", 1))),
+            submissionQuery.create(context, submissionMissingExecutionKeys),
+            submissionQuery.create(context, submissionWithExecutionKeys)
+          )
+        },
+        // update workflowExternalIdWithExecutionKey with an execution key
+        workflowQuery.findWorkflowByExternalId(workflowExternalIdWithExecutionKey).result flatMap {wr =>
+          workflowQuery.batchUpdateStatusAndExecutionServiceKey(wr, WorkflowStatuses.Submitted, ExecutionServiceId(instanceKeyForTests))
+        }
+      )
+    }
+  }
+
+  "ShardedHttpExecutionServiceCluster" should "return default instance" in {
+    val defaultEntry = cluster invokePrivate getDefaultEntry()
+    assertResult(defaultInstanceKey){ defaultEntry._1.id }
+  }
+
+  "a WorkflowRecord with an executionServiceKey" should "return the correct instance" in withCustomTestDatabase(execClusterTestData) {  dataSource: SlickDataSource =>
```

Comment:
all our other tests structure the test name like
it should "return the correct instance for a WorkflowRecord with an executionServiceKey"

this makes the test output look like all the tests falling under the ShardedHttpExecutionServiceCluster topic